File: conecta/src/main/java/com/sevenine/conecta/controller/DominioController.java
package com.sevenine.conecta.controller;

import com.sevenine.conecta.usecase.DominioUseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RequestMapping(path = "dominios")
@RestController
public class DominioController {

    private final DominioUseCase dominioUseCase;

    @GetMapping("genero")
    public ResponseEntity<?> generos() {
        return ResponseEntity.ok().body(dominioUseCase.consultarGeneros());
    }

    @GetMapping("tipo-empresa")
    public ResponseEntity<?> tiposEmpresa() {
        return ResponseEntity.ok().body(dominioUseCase.consultarTiposEmpresas());
    }

    @GetMapping("estado-civil")
    public ResponseEntity<?> estadosCivil() {
        return ResponseEntity.ok().body(dominioUseCase.consultarEstadosCivis());
    }

    @GetMapping("estado")
    public ResponseEntity<?> estados() {
        return ResponseEntity.ok().body(dominioUseCase.consultarEstados());
    }

    @GetMapping("conselho")
    public ResponseEntity<?> conselhos() {
        return ResponseEntity.ok().body(dominioUseCase.consultarConselhos());
    }

    @GetMapping("area-atendimento")
    public ResponseEntity<?> areasAtendimento() {
        return ResponseEntity.ok().body(dominioUseCase.consultarAreasAtendimentos());
    }

    @GetMapping("transporte")
    public ResponseEntity<?> transportes() {
        return ResponseEntity.ok().body(dominioUseCase.consultarTransportes());
    }

    @GetMapping("instrucao")
    public ResponseEntity<?> instrucoes() {
        return ResponseEntity.ok().body(dominioUseCase.consultarInstrucoes());
    }

    @GetMapping("categoria-cnh")
    public ResponseEntity<?> categoriasCnh() {
        return ResponseEntity.ok().body(dominioUseCase.consultarCategoriasCnh());
    }

    @GetMapping("tipo-conta")
    public ResponseEntity<?> tiposConta() {
        return ResponseEntity.ok().body(dominioUseCase.consultarTiposContas());
    }

    @GetMapping("banco")
    public ResponseEntity<?> bancos() {
        return ResponseEntity.ok().body(dominioUseCase.consultarBancos());
    }

    @GetMapping("pais")
    public ResponseEntity<?> paises() {
        return ResponseEntity.ok().body(dominioUseCase.consultarPaises());
    }

    @GetMapping("parentesco")
    public ResponseEntity<?> parentescos() {
        return ResponseEntity.ok().body(dominioUseCase.consultarParentescos());
    }

    @GetMapping("modelo")
    public ResponseEntity<?> modelos() {
        return ResponseEntity.ok().body(dominioUseCase.consultarModelos());
    }

    @GetMapping("grupo")
    public ResponseEntity<?> grupos() {
        return ResponseEntity.ok().body(dominioUseCase.consultarGrupos());
    }

    @GetMapping("status-atendimento")
    public ResponseEntity<?> status() {
        return ResponseEntity.ok().body(dominioUseCase.consultarStatusAtendimentos());
    }

}

File: service-conecta/src/main/java/com/sevenine/conecta/adapter/input/http/DominioController.kt
package com.sevenine.conecta.adapter.input.http

import com.sevenine.conecta.adapter.output.persistence.*
import com.sevenine.conecta.domain.model.*
import com.sevenine.conecta.domain.port.service.DominioServicePort
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("dominios")
@RestController
class DominioController(private val service: DominioServicePort) {

    @GetMapping("especialidade")
    fun especialidades(): ResponseEntity<List<Especialidade>> = ResponseEntity.ok(service
            .listaEspecialidades())

    @GetMapping("genero")
    fun generos(): ResponseEntity<List<Genero>> = ResponseEntity.ok(service
            .listaGeneros())

    @GetMapping("banco")
    fun bancos(): ResponseEntity<List<Banco>> = ResponseEntity.ok(service
            .listaBancos())

    @GetMapping("categoria-cnh")
    fun categoriasCnh(): ResponseEntity<List<CategoriaCnh>> = ResponseEntity.ok(service
            .listaCategoriasCnh())

    @GetMapping("conselho")
    fun conselhos(): ResponseEntity<List<Conselho>> = ResponseEntity.ok(service
            .listaConselhos())

    @GetMapping("estado")
    fun estados(): ResponseEntity<List<Estado>> = ResponseEntity.ok(service
            .listaEstados())

    @GetMapping("estado-civil")
    fun estadosCivil(): ResponseEntity<List<EstadoCivil>> = ResponseEntity.ok(service
            .listaEstadosCivis())

    @GetMapping("grupo")
    fun grupos(): ResponseEntity<List<Grupo>> = ResponseEntity.ok(service
            .listaGrupos())

    @GetMapping("instrucao")
    fun instrucoes(): ResponseEntity<List<Instrucao>> = ResponseEntity.ok(service
            .listaInstrucoes())

    @GetMapping("modelo")
    fun modelos(): ResponseEntity<List<Modelo>> = ResponseEntity.ok(service
            .listaModelos())

    @GetMapping("pais")
    fun paises(): ResponseEntity<List<Pais>> = ResponseEntity.ok(service
            .listaPaises())

    @GetMapping("parentesco")
    fun parentescos(): ResponseEntity<List<Parentesco>> = ResponseEntity.ok(service
            .listaParentescos())

    @GetMapping("tipo-conta")
    fun tiposConta(): ResponseEntity<List<TipoConta>> = ResponseEntity.ok(service
            .listaTiposConta())

    @GetMapping("tipo-empresa")
    fun tiposEmpresa(): ResponseEntity<List<TipoEmpresa>> = ResponseEntity.ok(service
            .listaTiposEsmpresa())

    @GetMapping("transporte")
    fun transportes(): ResponseEntity<List<Transporte>> = ResponseEntity.ok(service
            .listaTransportes())

}